### DISCUSSION TITLE
Can the operation_id of openapi be generated automatically?

### DISCUSSION THREAD

**Discussion by gudaoxuri** [poem-web/poem#523] [Ideas]
Some tools rely on operation_id to locate the interface. Writing operation_id manually is troublesome and error-prone. Can the operation_id of openapi be generated automatically?

![image](https://user-images.githubusercontent.com/2802567/222650740-4fc759d2-82d8-4fdf-aa5e-42ae5fbac6b0.png)

**Comment by VersBinarii** [comment]
You could most likely use build.rs to inject the id into some placeholder.
